optimum/onnxruntime/io_binding/io_binding_helper.py

```diff
@@ -48,7 +48,7 @@ def ort_type_to_numpy_type(ort_type: str):
             "tensor(int8)": np.int8,
             "tensor(float)": np.float32,
             "tensor(float16)": np.float16,
-            "tensor(bool)": np.uint8,
+            "tensor(bool)": bool,
```

Comment:
can we keep the numpy 1.24 fix in a separate PR? There are other issues still in ORT and tf2onnx.

Reply:
Actually, this is not related to 1.24? Why was it np.uint8 before?

Reply:
Only dlpack doesn't support bool as dtype(case for custom inputs when onnxruntime-training is installed). Previously I put it as `np.uint8` to make it compatible with the harsh constraint. But so far only with this PR we could have bool type input, so I change it to bool to support most common cases.